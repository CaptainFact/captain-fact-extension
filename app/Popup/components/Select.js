import React from 'react'
import classnames from 'classnames'
import { button, active } from '../../Common/components/Button.css'
import { select } from './Select.css'
import titleCase from 'voca/title_case'


export default class Select extends React.Component {
  render() {
    const { selected, options, onChange } = this.props
    return (
      <div className={select}>
        {Object.keys(options).map(key =>
          <a  key={key}
              className={classnames(button, {[active]: selected === options[key]})}
              onClick={() => selected !== options[key] && onChange(options[key])}>
            {titleCase(key)}
          </a>
        )}
      </div>
    );
  }
}
