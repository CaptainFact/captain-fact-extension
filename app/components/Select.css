.select > *:not(:first-child):not(:last-child) {
    border-radius: 0;
}

.select > *:not(:first-child) {
    margin-left: -1px;
}

.select > *:first-child {
    border-radius: 3px 0 0 3px;
}

.select > *:last-child {
    border-radius: 0 3px 3px 0;
}