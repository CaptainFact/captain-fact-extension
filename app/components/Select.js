import React from 'react'
import classnames from 'classnames'
import buttonStyles from './Button.css'
import selectStyles from './Select.css'

export default class Select extends React.Component {
  render() {
    const { selected, options, onChange } = this.props
    return (
      <div className={selectStyles.select}>
        {Object.keys(options).map((key) => (
          <a
            key={key}
            className={classnames(buttonStyles.button, {
              [buttonStyles.active]: selected === options[key],
            })}
            data-value={key}
            onClick={() => selected !== options[key] && onChange(options[key])}
          >
            {key}
          </a>
        ))}
      </div>
    )
  }
}
