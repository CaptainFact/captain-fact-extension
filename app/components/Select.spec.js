import React from 'react'
import renderer from 'react-test-renderer'
import Select from './Select.js'

describe('ON/OFF options', () => {
  const options = { ON: true, OFF: false }

  test('render with 1st option selected', () => {
    expect(
      renderer
        .create(<Select name="videosOverlay" selected options={options} />)
        .toJSON()
    ).toMatchInlineSnapshot(`
      <div
        className="select"
      >
        <a
          className="button active"
          data-value="ON"
          onClick={[Function]}
        >
          ON
        </a>
        <a
          className="button"
          data-value="OFF"
          onClick={[Function]}
        >
          OFF
        </a>
      </div>
    `)
  })

  test('render with second option selected', () => {
    expect(
      renderer
        .create(
          <Select name="videosOverlay" selected={false} options={options} />
        )
        .toJSON()
    ).toMatchInlineSnapshot(`
      <div
        className="select"
      >
        <a
          className="button"
          data-value="ON"
          onClick={[Function]}
        >
          ON
        </a>
        <a
          className="button active"
          data-value="OFF"
          onClick={[Function]}
        >
          OFF
        </a>
      </div>
    `)
  })
})
