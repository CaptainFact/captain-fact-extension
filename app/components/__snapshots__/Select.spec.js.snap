// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render selected option 1`] = `
<div
  className="select"
>
  <a
    className="button active"
    data-value="ON"
    key="ON"
    onClick={[Function]}
  >
    ON
  </a>
  <a
    className="button"
    data-value="OFF"
    key="OFF"
    onClick={[Function]}
  >
    OFF
  </a>
</div>
`;

exports[`render selected option 2`] = `
<div
  className="select"
>
  <a
    className="button"
    data-value="ON"
    key="ON"
    onClick={[Function]}
  >
    ON
  </a>
  <a
    className="button active"
    data-value="OFF"
    key="OFF"
    onClick={[Function]}
  >
    OFF
  </a>
</div>
`;
